.circleci/generate-ci.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+import os, sys
+from pprint import pprint
+from pathlib import Path
+
+if os.environ.get("CIRCLECI"):
+  os.system("sudo apt-get install python3-ruamel.yaml")
+from ruamel.yaml import YAML
+
+yaml = YAML(typ="rt")
+config = yaml.load(Path("config.pre.yml"))
+build_sink_deps = []
+
+def add_build_job(base_job, properties):
+  properties["context"] = ["reachdevbot-aws-ecr"]
+  config["workflows"]["build"]["jobs"].append({ base_job: properties })
+
+def dependencies(deps):
+  return {
+    "deps": " ".join(deps),
+    "requires": list(map(lambda name: f"build/{name}", deps))
+  }
+
+def build_image(executor, name, deps):
+  add_build_job("build-image", {
+    "name": f"build/{name}",
+    "image": name,
+    "exec": executor,
+    **dependencies(deps)
+  })
+
+def hs_test(mode):
+  name = f"hs-test-{mode}"
+  add_build_job("hs-test", {
+    "name": name,
+    "mode": mode,
+    **dependencies([f"haskell-build-artifacts-{mode}"])
+  })
+  build_sink_deps.append(name)
+
+def connector(executor, connector, parallelism):
+  examples_name = f"examples.{connector}"
+  browser_tests_name = f"browser-tests.{connector}"
+  devnet_name = f"devnet-{connector.lower()}"
+
+  build_image(executor, devnet_name, [])
+  add_build_job("examples", {
+    "name": examples_name,
+    "connector": connector,
+    "size": parallelism,
```

Comment:
This is not the way the other one worked. The other one did `total / 16` and you are doing `16`. This is very different.